activesupport/lib/active_support/message_verifier.rb

```diff
@@ -64,54 +65,96 @@ def valid_message?(signed_message)
     #
     # Returns +nil+ if the message is not Base64-encoded.
     #
-    #   invalid_message = "f--46a0120593880c733a53b6dad75b42ddc1c8996d"
+    #   invalid_message = "header.claims.MTkwMmI3MzY0MDA4NDM2MzVmYjgwMGEzOTJjOGQ2ZDM2Y2VhYjZiYg=="
     #   verifier.verified(invalid_message) # => nil
     #
     # Raises any error raised while decoding the signed message.
     #
-    #   incompatible_message = "test--dad7b06c94abba8d46a15fafaef56c327665d5ff"
+    #   incompatible_message = "aGVhZGVy.Y2xhaW1z.OWYzY2Y1Mzc1MmUxNjJlMzFhZDY3ODVlNGMyMTQ5YWE2OTc0NjQ5MQ=="
     #   verifier.verified(incompatible_message) # => TypeError: incompatible marshal file format
     def verified(signed_message)
-      if valid_message?(signed_message)
-        begin
-          data = signed_message.split("--")[0]
-          @serializer.load(decode(data))
-        rescue ArgumentError => argument_error
-          return if argument_error.message =~ %r{invalid base64}
-          raise
-        end
-      end
+      return unless valid_message?(signed_message)
+      
+      data = signed_message.split('.')[1]
+      @serializer.load(decode(data)).symbolize_keys
+    rescue ArgumentError => argument_error
+      raise unless argument_error.message =~ %r{invalid base64}
     end
 
-    # Decodes the signed message using the +MessageVerifier+'s secret.
+    # Decodes the signed message and legacy signed message using the +MessageVerifier+'s secret.
```

Comment:
I've said this a few times before, but please do take a look at how Global ID's signed version does this. The README explains the reasoning behind setting purpose and expiration: https://github.com/rails/globalid#signed-global-ids